### DISCUSSION TITLE
How do I debug the tests?

### DISCUSSION THREAD

**Discussion by diminutivesloop** [skills/secure-code-game#81] [Q&A] [answered]
I tried setting up debugging via the VSCode extension with the following config:
```
  "python.testing.unittestArgs": [
    "-v",
    "-s",
    "./Season-1/Level-1",
    "-p",
    "hack.py",
  ],
  "python.testing.pytestEnabled": false,
  "python.testing.unittestEnabled": true,
```
I can run the tests and they pass or fail as expected, but when I try to debug the tests I get a module error like this:
<img width="944" alt="Screenshot 2024-03-16 at 2 44 32 PM" src="https://github.com/skills/secure-code-game/assets/667716/684e688d-bf30-4f6a-aefa-234da6c086f9">

Is there a good way to resolve this or is there an alternate way to debug tests I can try?

**Comment by jkcso** [comment, marked as answer]
Hey @diminutivesloop, I've never tried to debug the tests or heard about anyone trying to do so so no experience there.
May I ask you why you want to debug the tests? To pass to the next level you need to change the `code` file so that both the `tests` and `hack` files pass.

**Reply by diminutivesloop** [reply]
By "tests" I primarily mean the `hack` files since they are also implemented as unit tests. I ended up peeking at the solution for season 1 level 1 but it would have been helpful to be able to step through the hack code to figure out why my solution wasn't working.

**Reply by jkcso** [reply]
Thanks for your feedback and will definitely have this in mind in future editions!
